autonomi/src/client/data_types/graph.rs

```diff
@@ -42,25 +43,29 @@ pub enum GraphError {
     InvalidQuote,
     #[error("Entry already exists at this address: {0:?}")]
     AlreadyExists(GraphEntryAddress),
+    #[error("Graph forked! Multiple entries found at this address: {0:?}")]
+    Fork(Vec<GraphEntry>),
 }
 
 impl Client {
     /// Fetches a GraphEntry from the network.
     pub async fn graph_entry_get(
         &self,
         address: GraphEntryAddress,
-    ) -> Result<Vec<GraphEntry>, GraphError> {
+    ) -> Result<GraphEntry, GraphError> {
         let graph_entries = self.network.get_graph_entry(address).await?;
-
-        Ok(graph_entries)
+        match &graph_entries[..] {
+            [entry] => Ok(entry.clone()),
+            multiple => Err(GraphError::Fork(multiple.to_vec())),
+        }
```

Comment:
Bit convoluted, but thought it could be done without cloning:

```suggestion
        match TryInto::<[_; 1]>::try_into(graph_entries) {
            Ok([a]) => Ok(a),
            Err(vec) => Err(GraphError::Fork(vec)),
        }
```

Reply:
Oh wow! Gonna keep it simple for now if you don't mind as _premature optimization is the root of all evil_!